openscap_report/scap_results_parser/parsers/oval_test_parser.py

```diff
@@ -78,43 +78,59 @@ def _get_variables_by_id(self):
         return self._get_data_by_id(data)
 
     @staticmethod
-    def _iter_over_data_and_get_references(dict_, out):
+    def _iter_over_data_and_get_references(dict_, out, data_id, map_referenced_oval_endpoints):
         for key, value in dict_.items():
             if isinstance(value, dict):
-                OVALTestParser._iter_over_data_and_get_references(value, out)
+                OVALTestParser._iter_over_data_and_get_references(
+                    value, out, data_id, map_referenced_oval_endpoints
+                )
             else:
                 matches_key = ["object_reference", "var_ref", "object_ref", "filter"]
                 matches_val = [":var:", ":obj:", ":ste:"]
                 if any(s in key for s in matches_key) and any(s in value for s in matches_val):
                     out.append(value)
+                    if data_id not in map_referenced_oval_endpoints:
+                        map_referenced_oval_endpoints[data_id] = []
+                    if value not in map_referenced_oval_endpoints[data_id]:
+                        map_referenced_oval_endpoints[data_id].append(value)
```

Comment:
Can the map be a collections.defaultdict? I think that would help you avoid this construction.